src/OrbitGl/LineGraphTrack.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "LineGraphTrack.h"
+
+#include <GteVector.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+
+#include "Geometry.h"
+#include "TextRenderer.h"
+#include "TimeGraph.h"
+#include "TimeGraphLayout.h"
+
+namespace orbit_gl {
+
+template <size_t Dimension>
+float LineGraphTrack<Dimension>::GetLabelYFromValues(
+    const std::array<double, Dimension>& values) const {
+  float content_height = this->size_[1] - this->layout_->GetTrackTabHeight() -
+                         this->layout_->GetTrackBottomMargin() - this->GetLegendHeight();
+  float base_y = this->pos_[1] - this->size_[1] + this->layout_->GetTrackBottomMargin();
+  double min = this->GetGraphMinValue();
+  double inverse_value_range = this->GetGraphInverseValueRange();
+  std::array<float, Dimension> normalized_values;
+  std::transform(values.begin(), values.end(), normalized_values.begin(),
+                 [min, inverse_value_range](double value) {
+                   return static_cast<float>((value - min) * inverse_value_range);
+                 });
+
+  if (Dimension == 1) return base_y + normalized_values[0] * content_height;
+  return base_y + content_height / 2.0f;
+}
+
+template <size_t Dimension>
+void LineGraphTrack<Dimension>::DrawSeries(Batcher* batcher, uint64_t min_tick, uint64_t max_tick,
+                                           float z) {
+  auto entries_affected_range_result =
+      this->series_.GetEntriesAffectedByTimeRange(min_tick, max_tick);
```

Comment:
In general, I am not sure whether we should store the data in the track or not.

Reply:
Are you talking about the place to store `series_`? 

The old `GraphTrack` has a `flat_hash_map<uint64_t, double> values_` which maps the timestamp to value. And this is a 1 dimensional time series. 
In the new `GraphTrack` we extend the 1 dimensional `values_` to the multi-dimensional `MultivariateTimeSeries`. 

I'm thinking to use the `MultivariateTimeSeries` instead of `TimerInfo` to store information for the graph tracks. So I'm kind of preferring keeping the time series data in the graph track.

Reply:
Yeah that was the direction of the question. I think for now let's keep it as it is. And as Pablo stated, refactor later to have the query.